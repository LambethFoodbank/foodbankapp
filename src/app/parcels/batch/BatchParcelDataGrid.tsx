"use client";

import { Button } from "@mui/material";
import { DataGrid, GridColDef } from "@mui/x-data-grid";
import React from "react";

export interface BatchGridDisplayRow {
    id: string | 0;
    rowName: number;
    fullName: string;
    phoneNumber: string;
    address: string;
    adults: number | null;
    children: number | null;
    listType: string;
    dietaryRequirements: string;
    feminineProducts: string;
    babyProducts: string;
    petFood: string;
    otherItems: string;
    deliveryInstructions: string;
    extraInformation: string;
    attentionFlag: string;
    signpostingCall: string;
    notes: string;
    voucherNumber: string;
    packingDate: string;
    packingSlot: string;
    shippingMethod: string;
    collectionInfo: string;
}

export const batchGridDisplayColumns: GridColDef<BatchGridDisplayRow[][number]>[] = [
    {
        field: "rowName",
        headerName: "Row Name",
        width: 150,
        editable: false,
        renderCell: (params) => {
            if (params.row.id === 0 && params.field === "rowName") {
                return <Button variant="contained">Apply Column</Button>;
            }
            return params.row.rowName;
        },
    },
    {
        field: "fullName",
        headerName: "Full Name",
        width: 150,
        editable: true,
    },
    {
        field: "phoneNumber",
        headerName: "Phone Number",
        width: 150,
        editable: true,
    },
    {
        field: "address",
        headerName: "Address",
        width: 200,
        editable: true,
    },
    {
        field: "adults",
        headerName: "Adults",
        type: "number",
        width: 110,
        editable: true,
    },
    {
        field: "children",
        headerName: "Children",
        type: "number",
        width: 110,
        editable: true,
    },
    {
        field: "listType",
        headerName: "List Type",
        width: 150,
        editable: true,
    },
    {
        field: "dietaryRequirements",
        headerName: "Dietary Requirements",
        width: 200,
        editable: true,
    },
    {
        field: "feminineProducts",
        headerName: "Feminine Products",
        width: 150,
        editable: true,
    },
    {
        field: "babyProducts",
        headerName: "Baby Products",
        width: 150,
        editable: true,
    },
    {
        field: "petFood",
        headerName: "Pet Food",
        width: 110,
        editable: true,
    },
    {
        field: "otherItems",
        headerName: "Other Items",
        width: 200,
        editable: true,
    },
    {
        field: "deliveryInstructions",
        headerName: "Delivery Instructions",
        width: 200,
        editable: true,
    },
    {
        field: "extraInformation",
        headerName: "Extra Information",
        width: 200,
        editable: true,
    },

    {
        field: "attentionFlag",
        headerName: "Attention Flag",
        width: 150,
        editable: true,
    },
    {
        field: "signpostingCall",
        headerName: "Signposting Call",
        width: 150,
        editable: true,
    },
    {
        field: "notes",
        headerName: "Notes",
        width: 200,
        editable: true,
    },
    {
        field: "voucherNumber",
        headerName: "Voucher Number",
        width: 150,
        editable: true,
    },
    {
        field: "packingDate",
        headerName: "Packing Date",
        width: 150,
        editable: true,
    },
    {
        field: "packingSlot",
        headerName: "Packing Slot",
        width: 150,
        editable: true,
    },
    {
        field: "shippingMethod",
        headerName: "Shipping Method",
        width: 150,
        editable: true,
    },
    {
        field: "collectionInfo",
        headerName: "Collection Info",
        width: 200,
        editable: true,
    },
];

const styledBatchGridDisplayColumns: GridColDef<BatchGridDisplayRow[][number]>[] =
    batchGridDisplayColumns.map((column) => {
        return { ...column, headerAlign: "center", align: "center" };
    });

interface BatchParcelDataGridProps {
    rows: BatchGridDisplayRow[];
}

export const defaultData: BatchGridDisplayRow[] = Array.from({ length: 2 }, (_, index) => ({
    id: crypto.randomUUID(),
    rowName: index,
    fullName: "",
    phoneNumber: "",
    address: "",
    adults: null,
    children: null,
    listType: "",
    dietaryRequirements: "",
    feminineProducts: "",
    babyProducts: "",
    petFood: "",
    otherItems: "",
    deliveryInstructions: "",
    extraInformation: "",
    attentionFlag: "",
    signpostingCall: "",
    notes: "",
    voucherNumber: "",
    packingDate: "",
    packingSlot: "",
    shippingMethod: "",
    collectionInfo: "",
}));
defaultData[0].id = 0;

const BatchParcelDataGrid: React.FC<BatchParcelDataGridProps> = ({ rows }) => {
    return (
        <DataGrid
            rows={rows}
            columns={styledBatchGridDisplayColumns}
            sx={{
                "& .MuiDataGrid-cell": {
                    border: "1px solid",
                    borderColor: "rgba(81, 81, 81, 0.5)",
                },
                "& .MuiDataGrid-columnHeaders": {
                    borderBottom: "3px solid",
                    borderTop: "2px solid",
                    borderColor: "rgba(81, 81, 81, 0.9)",
                },
                "& .MuiDataGrid-columnHeader": {
                    borderLeft: "1px solid",
                    borderRight: "1px solid",
                    borderColor: "rgba(81, 81, 81, 0.5)",
                },
                "& .MuiDataGrid-columnHeaderTitle": {
                    fontWeight: "bold",
                },
                border: "1px solid",
                borderColor: "rgba(81, 81, 81, 0.9)",
                margin: "1rem",
            }}
        />
    );
};

export default BatchParcelDataGrid;
