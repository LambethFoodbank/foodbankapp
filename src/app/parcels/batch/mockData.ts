import { Schema } from "@/databaseUtils";
import { BatchTableDataState } from "@/app/parcels/batch/BatchTypes";
import { BatchGridDisplayRow } from "@/app/parcels/batch/BatchParcelDataGrid";
import dayjs from "dayjs";

export const mockDisplayData: BatchGridDisplayRow[] = [
    {
        id: 0,
        rowName: 0,
        fullName: "",
        phoneNumber: "",
        address: "",
        adults: null,
        children: null,
        listType: "",
        dietaryRequirements: "",
        feminineProducts: "",
        babyProducts: "",
        petFood: "",
        otherItems: "",
        deliveryInstructions: "",
        extraInformation: "",
        attentionFlag: "",
        signpostingCall: "",
        notes: "",
        voucherNumber: "",
        packingDate: "",
        packingSlot: "",
        shippingMethod: "",
        collectionInfo: "",
    },
    {
        id: 1,
        rowName: 1,
        fullName: "John Doe",
        phoneNumber: "0123456789",
        address: "123 Main Street, Anytown, USA",
        adults: 3,
        children: 1,
        listType: "Hotel",
        dietaryRequirements: "Vegan",
        feminineProducts: "No",
        babyProducts: "Yes",
        petFood: "No",
        otherItems: "None",
        deliveryInstructions: "Leave at 10 am",
        extraInformation: "No special requests",
        attentionFlag: "Yes",
        signpostingCall: "No",
        notes: "None",
        voucherNumber: "123456789",
        packingDate: "2022-01-01",
        packingSlot: "Morning",
        shippingMethod: "Collection",
        collectionInfo: "Test centre, date, slot",
    },
    {
        id: 2,
        rowName: 2,
        fullName: "Jane Smiths",
        phoneNumber: "9876543210",
        address: "456 Elm Street, Anytown, USA",
        adults: 2,
        children: 1,
        listType: "Regular",
        dietaryRequirements: "Gluten-free",
        feminineProducts: "Yes",
        babyProducts: "No",
        petFood: "Yes",
        otherItems: "None",
        deliveryInstructions: "Leave at 12 pm",
        extraInformation: "No special requests",
        attentionFlag: "No",
        signpostingCall: "Yes",
        notes: "None",
        voucherNumber: "987654321",
        packingDate: "2022-01-02",
        packingSlot: "Afternoon",
        shippingMethod: "Delivery",
        collectionInfo: "Test centre, date, slot",
    },
];

export const mockTableDataState: BatchTableDataState = {
    overrideDataRow: {
        data: {
            client: {
                fullName: null,
                phoneNumber: null,
                address: null,
                adultInfo: null,
                childrenInfo: null,
                listType: null,
                dietaryRequirements: null,
                feminineProducts: null,
                babyProducts: null,
                nappySize: null,
                petFood: null,
                otherItems: null,
                deliveryInstructions: null,
                extraInformation: null,
                attentionFlag: null,
                signpostingCall: null,
                notes: null,
            },
            parcel: {
                voucherNumber: null,
                packingDate: null,
                packingSlot: null,
                shippingMethod: null,
                collectionInfo: null,
            },
        },
    },
    batchDataRows: [
        {
            id: 1,
            clientId: "1",
            data: {
                client: {
                    fullName: "John Doe",
                    phoneNumber: "0123456789",
                    address: {
                        addressLine1: "123 Main St",
                        addressLine2: null,
                        addressTown: "Anytown",
                        addressCounty: "USA",
                        addressPostcode: "12345",
                    },
                    adultInfo: {
                        adults: [],
                        numberOfAdults: 0,
                    },
                    childrenInfo: {
                        children: [],
                        numberOfChildren: 0,
                    },
                    listType: "hotel",
                    dietaryRequirements: { vegan: true },
                    feminineProducts: { tampons: false, pads: false },
                    babyProducts: null,
                    nappySize: null,
                    petFood: { cat: false, dog: false },
                    otherItems: { toothpaste: false, toothbrush: false },
                    deliveryInstructions: "Leave at 10 am",
                    extraInformation: "No special requests",
                    attentionFlag: true,
                    signpostingCall: false,
                    notes: null,
                },
                clientReadOnly: false,
                parcel: null,
            },
        },
        {
            id: 2,
            clientId: "2",
            data: {
                client: {
                    fullName: "Jane Smiths",
                    phoneNumber: "9876543210",
                    address: {
                        addressLine1: "456 Elm St",
                        addressLine2: null,
                        addressTown: "Anytown",
                        addressCounty: "USA",
                        addressPostcode: "54321",
                    },
                    adultInfo: {
                        adults: [],
                        numberOfAdults: 0,
                    },
                    childrenInfo: {
                        children: [],
                        numberOfChildren: 0,
                    },
                    listType: "regular",
                    dietaryRequirements: { "gluten-free": true },
                    feminineProducts: { tampons: true, pads: false },
                    babyProducts: false,
                    nappySize: null,
                    petFood: { cat: true, dog: false },
                    otherItems: { toothpaste: false, toothbrush: false },
                    deliveryInstructions: "Leave at 12 pm",
                    extraInformation: "No special requests",
                    attentionFlag: false,
                    signpostingCall: true,
                    notes: null,
                },
                clientReadOnly: false,
                parcel: null,
            },
        },
    ],
    clientOverrides: [],
    parcelOverrides: [],
};

export const mockExistingClientRow: Schema["clients"] = {
    primary_key: "dcb54bc0-b0d3-57fa-bf9b-f1c4da6931a9",
    full_name: "Test Person",
    phone_number: "+140969525594300",
    address_1: "3454 Test St",
    address_2: null,
    address_town: "Test Town",
    address_county: "Test County",
    address_postcode: "TE5T 1NG",
    delivery_instructions: "Leave at the door",
    family_id: "family-test-id",
    dietary_requirements: ["vegan"],
    feminine_products: ["tampons"],
    baby_food: false,
    pet_food: ["dog"],
    other_items: ["toothpaste"],
    extra_information: "No special requests",
    flagged_for_attention: false,
    signposting_call_required: true,
    last_updated: "2022-01-01",
    is_active: true,
    notes: null,
    default_list: "regular",
};

const currentYear = dayjs().year();
const currentMonth = dayjs().month();

export const mockExistingFamily: Schema["families"][] = [
    {
        primary_key: "test-dad",
        family_id: "family-test-id",
        gender: "male",
        birth_month: null,
        birth_year: 1980,
    },
    {
        primary_key: "test-mum",
        family_id: "family-test-id",
        gender: "female",
        birth_month: null,
        birth_year: 1985,
    },
    {
        primary_key: "test-adult-child",
        family_id: "family-test-id",
        gender: "female",
        birth_month: currentMonth,
        birth_year: currentYear - 16,
    },
    {
        primary_key: "test-child-child",
        family_id: "family-test-id",
        gender: "female",
        birth_month: currentMonth,
        birth_year: currentYear - 15,
    },
];

export const mockExistingClientRowWithNappySize: Schema["clients"] = {
    primary_key: "44d14896-357b-5e91-ab64-2ca95b6a2faa",
    full_name: "Test Nappy Person",
    phone_number: "+661129414051390",
    address_1: "3454 Nappy St",
    address_2: null,
    address_town: "Nappy Town",
    address_county: "Nappy County",
    address_postcode: "N4P 9YS",
    delivery_instructions: "Leave Nappies at the door",
    family_id: "nappy-family-test-id",
    dietary_requirements: ["vegan"],
    feminine_products: ["tampons"],
    baby_food: true,
    pet_food: ["dog"],
    other_items: ["toothpaste"],
    extra_information: "Nappy Size: 10, Extra Information: I love nappies",
    flagged_for_attention: false,
    signposting_call_required: true,
    last_updated: "2022-01-01",
    is_active: true,
    notes: null,
    default_list: "regular",
};

export const mockExistingNappyFamily: Schema["families"][] = [
    {
        primary_key: "test-nappy-dad",
        family_id: "nappy-family-test-id",
        gender: "male",
        birth_month: null,
        birth_year: 1980,
    },
    {
        primary_key: "test-nappy-baby",
        family_id: "nappy-family-test-id",
        gender: "female",
        birth_month: null,
        birth_year: currentYear - 2,
    },
];
