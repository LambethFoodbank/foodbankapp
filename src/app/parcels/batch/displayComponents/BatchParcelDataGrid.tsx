"use client";

import { Button, Dialog, DialogActions, DialogContent, DialogTitle } from "@mui/material";
import { DataGrid, GridColDef } from "@mui/x-data-grid";
import React, { useMemo, useState } from "react";
import { useLocalStorage, writeLocalTableState } from "@/app/parcels/batch/useLocalStorage";
import { BatchTableDataState } from "@/app/parcels/batch/batchTypes";
import batchParcelsReducer from "@/app/parcels/batch/batchParcelsReducer";
import { tableStateToBatchDisplayRows } from "@/app/parcels/batch/displayHelpers";
import { emptyBatchEditData, emptyOverrideData } from "@/app/parcels/batch/emptyData";
import { DefaultTheme, useTheme } from "styled-components";
import getCenteredBatchGridDisplayColumns from "@/app/parcels/batch/getCenteredBatchGridDisplayColumns";
import { useRouter } from "next/navigation";
import submitBatchTableData, {
    AddBatchRowError,
    filterUnsubmittedRows,
} from "@/app/parcels/batch/submitTableData";
import { batchSubmitTestData } from "@/app/parcels/batch/mockData";
import { verifyBatchTableData } from "@/app/parcels/batch/verifyTableData";

export interface BatchGridDisplayRow {
    [key: string]: string | number | null;
    id: number;
    fullName: string;
    phoneNumber: string;
    address: string;
    adults: number | null;
    children: number | null;
    listType: string;
    dietaryRequirements: string;
    feminineProducts: string;
    babyProducts: string;
    petFood: string;
    otherItems: string;
    deliveryInstructions: string;
    extraInformation: string;
    attentionFlag: string;
    signpostingCall: string;
    notes: string;
    voucherNumber: string;
    packingDate: string;
    packingSlot: string;
    shippingMethod: string;
    collectionInfo: string;
}

export const defaultTableState: BatchTableDataState = {
    overrideDataRow: {
        data: emptyOverrideData,
    },
    batchDataRows: [
        {
            id: 1,
            clientId: null,
            data: emptyBatchEditData,
        },
    ],
    clientOverrides: [],
    parcelOverrides: [],
};

const countErrorTypes = (errors: AddBatchRowError[]): Record<string, number> => {
    return errors.reduce(
        (acc, rowError) => {
            acc[rowError.displayMessage] = (acc[rowError.displayMessage] || 0) + 1;
            return acc;
        },
        {} as Record<string, number>
    );
};

const BatchParcelDataGrid: React.FC = () => {
    const [tableState, dispatch] = useLocalStorage(batchParcelsReducer, defaultTableState);
    const [isConfirmationDialogVisible, setIsConfirmationDialogVisible] = useState(false);
    const [isSuccessDialogVisible, setIsSuccessDialogVisible] = useState(false);
    const [dialogMessage, setDialogMessage] = useState("");
    const [confirmationErrors, setConfirmationErrors] = useState<AddBatchRowError[]>([]);
    const [submitErrors, setSubmitErrors] = useState<AddBatchRowError[]>([]);
    const router = useRouter();

    writeLocalTableState(batchSubmitTestData);

    const displayRows = useMemo(() => {
        return tableStateToBatchDisplayRows(tableState);
    }, [tableState]);

    const theme: DefaultTheme = useTheme();
    const [isRowCollection, setIsRowCollection] = useState<{ [key: number]: boolean }>({});
    const centeredBatchGridDisplayColumns: GridColDef[] = getCenteredBatchGridDisplayColumns(
        tableState,
        dispatch,
        isRowCollection,
        setIsRowCollection
    );

    const handleVerification = async (): Promise<void> => {
        const confirmationErrors = await verifyBatchTableData(tableState);
        setConfirmationErrors(confirmationErrors);
        setIsConfirmationDialogVisible(true);
    };

    const handleSubmit = async (): Promise<void> => {
        const { submitErrors } = await submitBatchTableData(tableState);
        setSubmitErrors(submitErrors);

        // comments will be updated and changed to be more specific in the next PR
        if (submitErrors.length == 0) {
            // setDialogMessage(`Successfully submitted ${tableState.batchDataRows.length} row(s)`);
            setDialogMessage("Successfully submitted these rows");
            handleReset();
        } else {
            // setDialogMessage(`There were ${submitErrors.length} error(s) encountered during submission.`);
            setDialogMessage("Errors encountered during submission.");
            handleUnsubmittedRows(submitErrors);
        }

        setIsConfirmationDialogVisible(false);
        setIsSuccessDialogVisible(true);
    };

    const handleReset = (): void => {
        setConfirmationErrors([]);
        setSubmitErrors([]);
        dispatch({
            type: "initialise_table_state",
            payload: { initialTableState: defaultTableState },
        });
    };

    const handleUnsubmittedRows = (submitErrors: AddBatchRowError[]): void => {
        const newState = filterUnsubmittedRows(tableState, submitErrors);
        dispatch({ type: "initialise_table_state", payload: { initialTableState: newState } });
    };

    const handleReturnToParcels = (): void => {
        router.push("/parcels");
    };

    return (
        <>
            <DataGrid
                rows={displayRows}
                columns={centeredBatchGridDisplayColumns}
                // only dispatches if a validated string field is updated
                processRowUpdate={(newRow: BatchGridDisplayRow, oldRow: BatchGridDisplayRow) => {
                    if (newRow.fullName !== oldRow.fullName) {
                        dispatch({
                            type: "update_cell",
                            updateCellPayload: {
                                rowId: newRow.id,
                                newValueAndFieldName: {
                                    type: "client",
                                    newValue: newRow.fullName,
                                    fieldName: "fullName",
                                },
                            },
                        });
                    }
                    if (newRow.voucherNumber !== oldRow.voucherNumber) {
                        dispatch({
                            type: "update_cell",
                            updateCellPayload: {
                                rowId: newRow.id,
                                newValueAndFieldName: {
                                    type: "parcel",
                                    newValue: newRow.voucherNumber,
                                    fieldName: "voucherNumber",
                                },
                            },
                        });
                    }
                    return newRow;
                }}
                // using sx rather than styled-components as otherwise throws an error when iterating through columns
                sx={{
                    "& .MuiDataGrid-cell": {
                        border: "1px solid",
                        borderColor: `${theme.main.border}`,
                        "& .MuiInputBase-root": {
                            height: "100%",
                        },
                    },
                    "& .MuiDataGrid-columnHeaders": {
                        borderBottom: "3px solid",
                        borderTop: "2px solid",
                        borderColor: `${theme.main.border}`,
                    },
                    "& .MuiDataGrid-columnHeader": {
                        borderLeft: "1px solid",
                        borderRight: "1px solid",
                        borderColor: `${theme.main.border}`,
                    },
                    "& .MuiDataGrid-columnHeaderTitle": {
                        fontWeight: "bold",
                    },
                    "& .Mui-error": {
                        backgroundColor: `${theme.error}`,
                        color: `${theme.text}`,
                    },
                    border: "1px solid",
                    borderColor: `${theme.main.border}`,
                    margin: "1rem",
                }}
                hideFooter
            />

            <Button
                onClick={handleVerification}
                variant="contained"
                sx={{ marginLeft: "1rem", minWidth: "120px" }}
            >
                Submit Data
            </Button>

            <Button
                onClick={() => dispatch({ type: "add_row" })}
                variant="contained"
                sx={{ marginLeft: "1rem", minWidth: "120px" }}
            >
                Add new row
            </Button>

            <Dialog
                open={isConfirmationDialogVisible}
                sx={{
                    "& .MuiDialog-paper": {
                        padding: "1rem",
                        borderRadius: "15px",
                        boxShadow: "0 3px 5px rgba(0,0,0,0.8)",
                        margin: "auto",
                        borderColor: confirmationErrors.length == 0 ? "#a8d49c" : "#990f0f",
                        borderWidth: "1px",
                        borderStyle: "solid",
                    },
                }}
            >
                <DialogTitle
                    sx={{
                        fontSize: "1.2rem",
                        textAlign: "center",
                    }}
                >
                    Are you sure you want to submit these rows?
                </DialogTitle>

                {confirmationErrors.length !== 0 && (
                    <DialogContent>
                        <ul>
                            {Object.entries(countErrorTypes(confirmationErrors)).map(
                                ([errorType, count]) => (
                                    <li key={errorType}>
                                        {errorType}: {count} row(s)
                                    </li>
                                )
                            )}
                        </ul>
                    </DialogContent>
                )}
                <DialogActions sx={{ justifyContent: "center" }}>
                    <>
                        <Button
                            variant="contained"
                            sx={{ margin: "1rem", minWidth: "120px" }}
                            onClick={() => {
                                setIsConfirmationDialogVisible(false);
                            }}
                        >
                            Close
                        </Button>
                        {confirmationErrors.length == 0 && (
                            <Button
                                variant="contained"
                                onClick={handleSubmit}
                                sx={{ margin: "1rem", minWidth: "120px" }}
                            >
                                Submit
                            </Button>
                        )}
                    </>
                </DialogActions>
            </Dialog>

            <Dialog
                open={isSuccessDialogVisible}
                sx={{
                    "& .MuiDialog-paper": {
                        padding: "1rem",
                        borderRadius: "15px",
                        boxShadow: "0 3px 5px rgba(0,0,0,0.8)",
                        margin: "auto",
                        borderColor: submitErrors.length == 0 ? "#a8d49c" : "#990f0f",
                        borderWidth: "1px",
                        borderStyle: "solid",
                    },
                }}
            >
                <DialogTitle
                    sx={{
                        fontSize: "1.2rem",
                        textAlign: "center",
                    }}
                >
                    {dialogMessage}
                </DialogTitle>
                {submitErrors.length > 0 && (
                    <DialogContent sx={{ textAlign: "center" }}>
                        An unexpected error occurred and some rows did not submit.
                        <br />
                        Please try submitting again.
                    </DialogContent>
                )}
                <DialogActions sx={{ justifyContent: "center" }}>
                    {submitErrors.length > 0 ? (
                        <>
                            <Button
                                variant="contained"
                                sx={{ margin: "1rem", minWidth: "120px" }}
                                onClick={() => {
                                    handleReset();
                                    setIsSuccessDialogVisible(false);
                                }}
                            >
                                Reset rows and start again
                            </Button>
                            <Button
                                variant="contained"
                                onClick={() => setIsSuccessDialogVisible(false)}
                                sx={{ margin: "1rem", minWidth: "120px" }}
                            >
                                View unsubmitted rows
                            </Button>
                        </>
                    ) : (
                        <>
                            <Button
                                variant="contained"
                                sx={{ margin: "1rem", minWidth: "120px" }}
                                onClick={() => setIsSuccessDialogVisible(false)}
                            >
                                Add more rows
                            </Button>
                            <Button
                                variant="contained"
                                sx={{ margin: "1rem", minWidth: "120px" }}
                                onClick={handleReturnToParcels}
                            >
                                Return to the Parcels Page
                            </Button>
                        </>
                    )}
                </DialogActions>
            </Dialog>
        </>
    );
};

export default BatchParcelDataGrid;
