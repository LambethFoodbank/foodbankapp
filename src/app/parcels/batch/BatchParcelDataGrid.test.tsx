import BatchParcelDataGrid from "@/app/parcels/batch/BatchParcelDataGrid";
import React from "react";
import { expect, it } from "@jest/globals";
import "@testing-library/jest-dom/jest-globals";
import renderer from "react-test-renderer";
import mockData from "./mockData";

describe("Data Grid Snapshot Test", () => {
    it("Matches DOM Snapshot", () => {
        const domTree = renderer
            .create(<BatchParcelDataGrid rows={mockData} />)
            .toJSON() as renderer.ReactTestRendererJSON;
        expect(domTree).toMatchSnapshot();
    });
});

export default mockData;
