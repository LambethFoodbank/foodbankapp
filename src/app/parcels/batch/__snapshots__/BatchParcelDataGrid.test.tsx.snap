// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Grid Snapshot Test Matches DOM Snapshot 1`] = `
<div
  aria-colcount={22}
  aria-multiselectable={false}
  aria-rowcount={4}
  className="MuiDataGrid-root MuiDataGrid-root--densityStandard MuiDataGrid-withBorderColor css-1r7o7si-MuiDataGrid-root"
  role="grid"
>
  <div
    className="MuiDataGrid-main css-204u17-MuiDataGrid-main"
  >
    <div
      className="MuiDataGrid-columnHeaders MuiDataGrid-withBorderColor css-1iyq7zh-MuiDataGrid-columnHeaders"
      role="presentation"
      style={
        {
          "lineHeight": "56px",
          "maxHeight": 56,
          "minHeight": 56,
        }
      }
    >
      <div
        className="MuiDataGrid-columnHeadersInner css-gl260s-MuiDataGrid-columnHeadersInner"
        role="rowgroup"
      />
    </div>
  </div>
  <div
    className="MuiDataGrid-footerContainer MuiDataGrid-withBorderColor css-wop1k0-MuiDataGrid-footerContainer"
  >
    <div />
    <div
      className="MuiTablePagination-root css-rtrcn9-MuiTablePagination-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
      >
        <div
          className="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
        />
        <p
          className="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
          id=":r1:"
        >
          Rows per page:
        </p>
        <div
          className="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-input css-16c50h-MuiInputBase-root-MuiTablePagination-select"
          onClick={[Function]}
        >
          <div
            aria-controls=":r2:"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby=":r1: :r0:"
            className="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
            id=":r0:"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="combobox"
            tabIndex={0}
          >
            100
          </div>
          <input
            aria-hidden={true}
            aria-invalid={false}
            className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            disabled={false}
            onAnimationStart={[Function]}
            onChange={[Function]}
            tabIndex={-1}
            value={100}
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <p
          className="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
        >
          1â€“3 of 3
        </p>
        <div
          className="MuiTablePagination-actions"
        >
          <button
            aria-label="Go to previous page"
            className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            title="Go to previous page"
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </button>
          <button
            aria-label="Go to next page"
            className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            title="Go to next page"
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
